### PR TITLE
Improve git log for debugging

### PR DESCRIPTION
**Pull request by wxiaoguang**
Help https://github.com/go-gitea/gitea/issues/22716#issuecomment-1506265149

### CODE REVIEW

File: modules/git/command.go [hunk 1]
```diff
@@ -49,10 +49,28 @@ type Command struct {
...
+	debugQuote := func(s string) string {
+		if strings.ContainsAny(s, " `'\"\t\r\n") {
+			return fmt.Sprintf("%q", s)
+		}
+		return s
+	}
+	a := make([]string, 0, len(c.args)+1)
+	a = append(a, debugQuote(c.prog))
+	for _, arg := range c.args {
+		if sanitizing && (strings.Contains(arg, "://") && strings.Contains(arg, "@")) {
```
Comment (delvh) [inline modules/git/command.go:67]: I think you mean
```suggestion
		if sanitizing && (strings.Contains(arg, "://") || strings.Contains(arg, "@")) {
```

File: modules/git/repo.go [hunk 2]
```diff
@@ -209,49 +209,22 @@ func Push(ctx context.Context, repoPath string, opts PushOptions) error {
 	} else {
 		cmd.SetDescription(fmt.Sprintf("push branch %s to %s (force: %t, mirror: %t)", opts.Branch, opts.Remote, opts.Force, opts.Mirror))
 	}
-	var outbuf, errbuf strings.Builder
 
-	if opts.Timeout == 0 {
-		opts.Timeout = -1
-	}
```
Comment (delvh) [inline modules/git/repo.go:216]: What happened to this check?

File: modules/git/command.go (hunk 1 above)
Comment (wxiaoguang) [reply modules/git/command.go:67]: `&&` is right

File: modules/git/repo.go (hunk 2 above)
Comment (wxiaoguang) [reply modules/git/repo.go:216]: `Timeout = 0` also means default timeout

Comment (wolfogre) [inline modules/git/repo.go:216]: I was worried about removing this, but it seems OK.

https://github.com/go-gitea/gitea/blob/6bc3079c0036a54d1b8ab04c5a6e14111c719c9a/modules/git/command.go#L248

**Review by wolfogre — APPROVED** [review]

File: modules/git/command.go (hunk 1 above)
Comment (delvh) [reply modules/git/command.go:67]: Why?
SSH doesn't have `://`…

Comment (wxiaoguang) [reply modules/git/command.go:67]: SSH without `://` doesn't have password either.

File: modules/git/repo.go (hunk 2 above)
Comment (wxiaoguang) [reply modules/git/repo.go:216]: Yup, this code ` if timeout <= 0 { ` was also written by me 🤣

**Review by delvh — APPROVED** [review]

**Pull request merged**